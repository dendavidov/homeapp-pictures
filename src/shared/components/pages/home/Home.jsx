import React from 'react';

import kitten from './assets/kitten.jpg';

import styles from './Home.styl';

class Home extends React.Component {
  state = {};

  render() {
    return (
      <div className={styles.root}>
        <h1>KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)</h1>
        <h2>About</h2>
        <p>
          This project is based on{' '}
          <a href="https://github.com/ctrlplusb/react-universally">
            React, Universally (A starter kit for universal react applications)
          </a>. It{"'"}s been made for my personal needs, nonetheless you may
          use it whatever you want.
        </p>
        <h2>Features</h2>
        <ul>
          <li>Koa - server;</li> <li>React - view;</li>
          <li>Jest - test framework;</li> <li>Stylus - CSS preprocessor;</li>
          <li>MongoDB - NoSQL database;</li>
          <li>Mongoose - Object Data Modeling (ODM) library for MongoDB.</li>
        </ul>
        <img src={kitten} alt="Kitten" />
      </div>
    );
  }
}

export default Home;
