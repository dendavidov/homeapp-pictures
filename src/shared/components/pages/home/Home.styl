.root
  width 100%

.wrapper
  display flex
  padding 10px 0

.photosContainer
  width 80%
  display flex
  flex-wrap wrap
  justify-content flex-start

.photo
  border none
  width 200px
  height 200px
  margin 0 10px 10px 10px
  background-size cover
  cursor pointer

.labelsContainer
  display flex
  flex-direction column
  flex 1

.label
  display flex
  justify-content space-between

.label
  div:last-child
    width 50%
