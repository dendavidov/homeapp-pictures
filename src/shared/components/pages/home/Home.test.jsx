import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';

describe('Home', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('should render', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it('should render div', () => {
    expect(wrapper.is('div')).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
