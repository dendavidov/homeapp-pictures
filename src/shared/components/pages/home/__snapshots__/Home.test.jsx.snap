// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)
        </h1>,
        <h2>
          About
        </h2>,
        <p>
          This project is based on
           
          <a
            href="https://github.com/ctrlplusb/react-universally"
          >
            React, Universally (A starter kit for universal react applications)
          </a>
          . It
          '
          s been made for my personal needs, nonetheless you may use it whatever you want.
        </p>,
        <h2>
          Features
        </h2>,
        <ul>
          <li>
            Koa - server;
          </li>
           
          <li>
            React - view;
          </li>
          <li>
            Jest - test framework;
          </li>
           
          <li>
            Stylus - CSS preprocessor;
          </li>
          <li>
            MongoDB - NoSQL database;
          </li>
          <li>
            Mongoose - Object Data Modeling (ODM) library for MongoDB.
          </li>
        </ul>,
        <img
          alt="Kitten"
          src="kitten.jpg"
        />,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)",
        },
        "ref": null,
        "rendered": "KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "About",
        },
        "ref": null,
        "rendered": "About",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "This project is based on",
            " ",
            <a
              href="https://github.com/ctrlplusb/react-universally"
            >
              React, Universally (A starter kit for universal react applications)
            </a>,
            ". It",
            "'",
            "s been made for my personal needs, nonetheless you may use it whatever you want.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "This project is based on",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "React, Universally (A starter kit for universal react applications)",
              "href": "https://github.com/ctrlplusb/react-universally",
            },
            "ref": null,
            "rendered": "React, Universally (A starter kit for universal react applications)",
            "type": "a",
          },
          ". It",
          "'",
          "s been made for my personal needs, nonetheless you may use it whatever you want.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Features",
        },
        "ref": null,
        "rendered": "Features",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Koa - server;
            </li>,
            " ",
            <li>
              React - view;
            </li>,
            <li>
              Jest - test framework;
            </li>,
            " ",
            <li>
              Stylus - CSS preprocessor;
            </li>,
            <li>
              MongoDB - NoSQL database;
            </li>,
            <li>
              Mongoose - Object Data Modeling (ODM) library for MongoDB.
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Koa - server;",
            },
            "ref": null,
            "rendered": "Koa - server;",
            "type": "li",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "React - view;",
            },
            "ref": null,
            "rendered": "React - view;",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Jest - test framework;",
            },
            "ref": null,
            "rendered": "Jest - test framework;",
            "type": "li",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Stylus - CSS preprocessor;",
            },
            "ref": null,
            "rendered": "Stylus - CSS preprocessor;",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MongoDB - NoSQL database;",
            },
            "ref": null,
            "rendered": "MongoDB - NoSQL database;",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Mongoose - Object Data Modeling (ODM) library for MongoDB.",
            },
            "ref": null,
            "rendered": "Mongoose - Object Data Modeling (ODM) library for MongoDB.",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Kitten",
          "src": "kitten.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)
          </h1>,
          <h2>
            About
          </h2>,
          <p>
            This project is based on
             
            <a
              href="https://github.com/ctrlplusb/react-universally"
            >
              React, Universally (A starter kit for universal react applications)
            </a>
            . It
            '
            s been made for my personal needs, nonetheless you may use it whatever you want.
          </p>,
          <h2>
            Features
          </h2>,
          <ul>
            <li>
              Koa - server;
            </li>
             
            <li>
              React - view;
            </li>
            <li>
              Jest - test framework;
            </li>
             
            <li>
              Stylus - CSS preprocessor;
            </li>
            <li>
              MongoDB - NoSQL database;
            </li>
            <li>
              Mongoose - Object Data Modeling (ODM) library for MongoDB.
            </li>
          </ul>,
          <img
            alt="Kitten"
            src="kitten.jpg"
          />,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)",
          },
          "ref": null,
          "rendered": "KOA-REACT-UNIVERSAL-BOILERPLATE (KRUMB)",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "About",
          },
          "ref": null,
          "rendered": "About",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "This project is based on",
              " ",
              <a
                href="https://github.com/ctrlplusb/react-universally"
              >
                React, Universally (A starter kit for universal react applications)
              </a>,
              ". It",
              "'",
              "s been made for my personal needs, nonetheless you may use it whatever you want.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "This project is based on",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "React, Universally (A starter kit for universal react applications)",
                "href": "https://github.com/ctrlplusb/react-universally",
              },
              "ref": null,
              "rendered": "React, Universally (A starter kit for universal react applications)",
              "type": "a",
            },
            ". It",
            "'",
            "s been made for my personal needs, nonetheless you may use it whatever you want.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Features",
          },
          "ref": null,
          "rendered": "Features",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Koa - server;
              </li>,
              " ",
              <li>
                React - view;
              </li>,
              <li>
                Jest - test framework;
              </li>,
              " ",
              <li>
                Stylus - CSS preprocessor;
              </li>,
              <li>
                MongoDB - NoSQL database;
              </li>,
              <li>
                Mongoose - Object Data Modeling (ODM) library for MongoDB.
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Koa - server;",
              },
              "ref": null,
              "rendered": "Koa - server;",
              "type": "li",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "React - view;",
              },
              "ref": null,
              "rendered": "React - view;",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jest - test framework;",
              },
              "ref": null,
              "rendered": "Jest - test framework;",
              "type": "li",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Stylus - CSS preprocessor;",
              },
              "ref": null,
              "rendered": "Stylus - CSS preprocessor;",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MongoDB - NoSQL database;",
              },
              "ref": null,
              "rendered": "MongoDB - NoSQL database;",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mongoose - Object Data Modeling (ODM) library for MongoDB.",
              },
              "ref": null,
              "rendered": "Mongoose - Object Data Modeling (ODM) library for MongoDB.",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Kitten",
            "src": "kitten.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
